src/vs/workbench/parts/terminal/electron-browser/terminalInstance.ts

```diff
@@ -688,6 +688,8 @@ export class TerminalInstance implements ITerminalInstance {
 						c(this._escapeNonWindowsPath(stdout.trim()));
 					});
 					return;
+				} else if (hasSpace && (exe.indexOf('powershell') !== -1)) {
```

Comment:
In the latest version of PowerShell, PowerShell Core (xplat, ships outside of Windows), the executable has been renamed to `pwsh` to prevent any conflicts with the PowerShell version that ships in Windows, called Windows PowerShell aka `powershell.exe`.

This fix should also include a check for the latest version of PowerShell as well.

```Javascript
exe.indexOf('powershell') !== -1 || exe.indexOf('pwsh') !== -1
```

Also, I think this fix only works on Windows, but PowerShell Core works on macOS and Linux.

cc @Tyriar I probably should have called out the executable name in the issue you tagged me in. My bad.

Reply:
@TylerLeonhardt 👍, you can do a PR for that to get credit for it if you want?